### DISCUSSION TITLE
Enhancement to better support LUNs and devices that look for them.

### DISCUSSION THREAD

**Discussion by dkgrizzly** [erichelgeson/BlueSCSI-v1#57] [Ideas]
Message 0x8? takes precedence over the LUN specified in the CDB.
One way to handle this is to set m_lun to an invalid placeholder value upon selection, change it when 0x8? is received, and only set it in the command parser if it is invalid still at that point.

Additionally, only RequestSense and Inquiry commands are required to work for an invalid LUN. All others can safely throw a Check Condition status.
onRequestSense can be modified thus:
`  if(!m_img) {`
`   buf[2] = 0x02; // NOT_READY`
`    buf[12] = 0x25; // Logical Unit Not Supported`
`  } else {`
`    ...`
`  }`

and in onInquiryCommand:
`  byte buf[36];`
 ` memcpy(buf, SCSI_INFO_BUF, 36);`
`  if(!m_img) buf[0] = 0x7f;`
`  writeDataPhase(len < 36 ? len : 36, buf);`

This makes things that scan for multiple LUNs much happier when there are not LUNs present.

**Comment by erichelgeson** [comment]
Hey thanks! I dont know why I dont get notified of Github Discussions, sorry for the late reply. I'll give this patch a try.
